src/pybind/mgr/diskprediction/agent/metrics/db_relay.py

```diff
@@ -0,0 +1,561 @@
+            data.tags['host'] = cluster_host
+            data.fields['cmd'] = str(cypher_cmd)
+            self._host_nodes[host] = host_node
+            self.data.append(data)
+
+    def _host_contains_mon(self):
+        cluster_id = self._cluster_id
+
+        db = DB_API(self._ceph_context)
+        mons = db.get_mons()
+        for mon in mons:
+            mon_name = mon.get('name', '')
+            mon_addr = mon.get('addr', '').split(':')[0]
+            for hostname in self._host_nodes:
+                if hostname != mon_name:
+                    continue
+
+                host_node = self._host_nodes[hostname]
+                data = DB_Relay()
+                dp_mon = MGRDpMon(
+                    fsid=cluster_id,
+                    host=mon_name,
+                    ipaddr=mon_addr
+                )
+
+                # create mon node
+                mon_node = NodeInfo(
+                    label='CephMon',
+                    domain_id="{}.mon.{}".format(cluster_id, mon_name),
+                    name=mon_name,
+                    meta=dp_mon.__dict__
+                )
+
+                # add mon node relationship
+                cypher_cmd = CypherOP.add_link(
+                    host_node,
+                    mon_node,
+                    'HostContainsMon'
+                )
+                cluster_host = socket.gethostname()
+                data.fields['agenthost'] = cluster_host
+                data.tags['agenthost_domain_id'] = \
+                    str("%s_%s" % (cluster_id, data.fields['agenthost']))
+                data.tags['host'] = cluster_host
+                data.fields['cmd'] = str(cypher_cmd)
+                self.data.append(data)
+
+    def _host_contains_osd(self):
+        cluster_id = self._cluster_id
+
+        db = DB_API(self._ceph_context)
+        osd_data = db.get_osd_data()
+        osd_journal = db.get_osd_journal()
+        for _data in db.get_osds():
+            osd_id = _data['osd']
+            osd_uuid = _data['uuid']
+            osd_up = _data['up']
+            osd_in = _data['in']
+            if not osd_in:
+                continue
+            osd_weight = _data['weight']
+            osd_public_addr = _data['public_addr']
+            osd_cluster_addr = _data['cluster_addr']
+            osd_heartbeat_back_addr = _data['heartbeat_back_addr']
+            osd_heartbeat_front_addr = _data['heartbeat_front_addr']
+            osd_state = _data['state']
+            osd_metadata = db.get_osd_metadata(osd_id)
+            if osd_metadata:
+                data = DB_Relay()
+                osd_host = osd_metadata['hostname']
+                osd_ceph_version = osd_metadata['ceph_version']
+                osd_rotational = osd_metadata['rotational']
+                osd_devices = osd_metadata['devices'].split(',')
+
+                # filter 'dm' device.
+                devices = []
+                for devname in osd_devices:
+                    if 'dm' in devname:

```

Comment:
This is dangerous and has unexpected behaviour if you are processing device mapper devs here. Use a regex instead to match exactly what you want, maybe something like ``(mapper\/.+)|(dm-\d+)``.